### PR TITLE
2.x and 3.0 compat with symfony/console

### PR DESCRIPTION
**Pull request by nicolas-grekas**
Replaces #214

### CODE REVIEW

**Comment by nicolas-grekas** [comment]
See also https://github.com/symfony/symfony/pull/15657
Even if this PR on symfony is merged, I think this one is right.

File: src/Psy/Shell.php [hunk 1]
```diff
@@ -279,7 +279,13 @@ public function run(InputInterface $input = null, OutputInterface $output = null
...
         }
 
-        return parent::run($input, $output);
+        try {
+            return parent::run($input, $output);
+        } catch (\Exception $e) {
+        } catch (\Throwable $e) {
+        }
+
+        $this->writeException($e);
```
Comment (GrahamCampbell) [inline src/Psy/Shell.php]: this could result in a fatal error

Comment (GrahamCampbell) [inline src/Psy/Shell.php]: you need to convert the throwable object to an exception first

Comment (nicolas-grekas) [inline src/Psy/Shell.php]: right, I've removed the cath for throwables, it's an other topic for an other PR

**Comment by bobthecow** [comment]
Thanks @nicolas-grekas!

**Pull request merged**